pkg/network/types.go

```diff
@@ -36,3 +47,40 @@ func (c CIDR) ToIPNet() *net.IPNet {
 	netw := net.IPNet(c)
 	return &netw
 }
+
+func (c *CIDR) IsIPv4() bool {
+	return c.IP.To4() != nil
+}
+
+// ParseIPNet parses a CIDR string and returns a network.CIDR (for user-provided values)
+func ParseIPNet(cidr string) (CIDR, error) {
+	_, prefix, err := net.ParseCIDR(cidr)
+	if err != nil {
+		return CIDR{}, err
+	}
+	return CIDR(*prefix), nil
+}
+
+// ParseIPNetIgnoreError parses a CIDR string and ignores any error (for testing and constants)
+func ParseIPNetIgnoreError(cidr string) CIDR {
+	parsed, _ := ParseIPNet(cidr)
+	return parsed
+}
+
+// GetByIPFamily returns a list of CIDRs that belong to the given IP family.
+func GetByIPFamily(cidrs []CIDR, ipFamily string) []CIDR {
+	var result []CIDR
+	for _, nw := range cidrs {
+		switch ipFamily {
+		case IPv4Family:
+			if nw.IP.To4() != nil && len(nw.IP) == net.IPv4len {
+				result = append(result, nw)
+			}
+		case IPv6Family:
+			if nw.IP.To16() != nil && len(nw.IP) == net.IPv6len {
```

Comment:
The current solution is not correct, but the proposed solution is also not strictly correct for incorrect values.

Usually, we just switch between the result of `To4()` meaning either IPv4 or IPv6.

```suggestion
			if nw.IP.To4() == nil {
```